### ISSUE TITLE
Checked checkboxes should have `checked="true"` and not `aria-checked="true"`

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2514]
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/trimble-oss/modus-web-components/blob/main/CONTRIBUTING.md)

### Describe the issue

The current behaviour is problematic.

- Checked checkboxes should have `checked="true"` (and not `aria-checked="true"`)
- Unchecked checkboxes should have `checked="false"` (and not `aria-checked="false"`)

Deque university has more details on this:
https://dequeuniversity.com/rules/axe/4.9/aria-conditional-attr?application=AxeEdge

### Reduced test cases

Run Axe Core scan on: https://coliff.github.io/modus-web-components-example/

![image](https://github.com/trimble-oss/modus-web-components/assets/1212885/3ccc032a-ec75-4ab1-87e5-fd943e473ac9)

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Microsoft Edge

### What is the issue regarding ?

@trimble-oss/modus-web-components

### What version of npm package are you using ?

latest

### Priority

Medium

### What product/project are you using Modus Components for ?

Modus

### What is your team/division name ?

Modus

### Are you willing to contribute ?

Yes
